### ISSUE TITLE
Improve styles for a11y and consistency across all themes

### ISSUE THREAD

**Issue by aristath** [WordPress/gutenberg#16782]
One of the main issues I have with Gutenberg is the styles specificity causing accessibility issues on the frontend. And by specificity here I don't mean the specificity of selectors. The main issue I have is with hardcoded color values and dimensions in pixels.
Most times these things result in accessibility issues on the frontend due to extremely small font-sizes, or insufficient color contrast.
I'll use the recent-comments block as an example, but the same thing applies to all blocks.
Currently we have styles like this:
```css
.wp-block-latest-comments__comment-date {
    color: $dark-gray-100;
    font-size: 12px;
}
```
With the above styles if a site happens to have a dark background, that text is unreadable because there's just not enough contrast.
Furthermore, `12px` is way too small for my ageing eyes, and using anything below 14px is just bad practice. It may have been ok on printed parer, but not on a 15'' laptop at a distance of 50cm... That's why a lot of sites have started switching to a base font-size of 18px or more.
The way Gutenberg styles are coded, themes are forced to override almost everything.
It would be a lot better if the above styles were instead written like this:
```css
.wp-block-latest-comments__comment-date {
    color: currentColor;
    opacity: .6;
    font-size: .85em;
}
```
And we don't even need the `currentColor` there since it will be inherited, the above is just an example of an override I had to put in my theme.
With an implementation like that, if the site has a white background the text color will be dark gray. If the site has a dark background (or if the user has their browser on dark mode for a11y reasons and the site takes that into account and respects user preferences) then the text color will be light gray. No matter what happens it will remain readable.
As for the font-size in the example, if the base font-size is 14px then the element's size will be approximately 12px, and if the base font-size is 18px we won't have an unreadable block of text. It will resize proportionately to match the rest of the site's styles

Gutenberg should not assume that all sites have a white background. It should not assume that all sites have a [base font-size of `13px`](https://github.com/WordPress/gutenberg/blob/5430dced1e3594ca28f5a9c829aa9baac8905e27/assets/stylesheets/_variables.scss#L7)
Using less specific and more agnostic styles would significantly improve accessibility without sacrificing quality, and at the same time it would allow themes to override less things in order to be accessible.

**Comment by aristath** [comment]
Closing this one. We keep improving the styles as we go on a per-block basis, and we keep revising things. With global-styles in the horizon, this no longer makes a lot of sense and should be handled as we come across issues.
